Luke/whitespace

lukesmurray opened the pull request:
This is to fix #438. I believe this works and is DRY. My only gripe is that I used `serialized + ""` in two places. If you want I can extract that into a variable.

LeaVerou commented on src/formats.js:
```diff
@@ -34,7 +34,8 @@ var base = _.Base = $.Class({
 	static: {
-		parse: serialized => Promise.resolve(serialized? JSON.parse(serialized) : null),
+		parse: serialized => Promise.resolve(serialized && (serialized + "").trim() ? 
+			JSON.parse((serialized + "").trim()) : null),
```
It is not, `JSON.parse()` converts to a string internally anyway. You can quickly verify this by typing e.g. `JSON.parse(false)` in the console. :)

lukesmurray commented on src/mavo.js:
```diff
@@ -474,15 +474,22 @@ var _ = self.Mavo = $.Class({
-			// Try again with init
+			// if there is an init backend then return null and don't propagate the error
 			if (this.init && this.init != backend) {
 				backend = this.init;
```
not sure if we need to change the backend variable here.

lukesmurray commented:
Sorry I committed early I still haven't fixed the issue :-1:

lukesmurray commented:
Ok so I think this fulfills what you were asking for but the error is still propagated. 
The part I'm confused about is what we want to do in `mavo#load ` when the data is null and there is no init. Do we want to suppress any errors in that case and assume that the backend is empty?

LeaVerou commented:
> The part I'm confused about is what we want to do in mavo#load when the data is null and there is no init. Do we want to suppress any errors in that case and assume that the backend is empty?

Yes. It's not an error, most apps start with no data! If there's an init backend we load from there, if not, no problem, we just render null!

LeaVerou commented:
Hi Luke,
Are you still interested in working on this? Do you need help?

Cheers,
Lea

lukesmurray commented:
@LeaVerou better late than never. This passes the tests you wrote. I did what you suggested.

1. removed the error I added for when the backend loads a null value.
2. return null from the first `catch` if the backend load throws an error and init exists
3. add a `then` which attempts to load with init if the data is null and init exists.
4. propagate null in the new `then` if init does not exist.
